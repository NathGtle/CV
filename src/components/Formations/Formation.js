import React from 'react'

const Formation = ({ formation }) => (
  <article className="formation">
    <div className="headline">{`${formation.nom} (${formation.lieu})`}</div>
    <div className="duration">
      {`${formation.dateDebut} ${formation.dateFin !== null ? `- ${formation.dateFin}` : ''}`}
    </div>
    <div className="clear" />
    <div className="description">{formation.description}</div>
  </article>
)

export default Formation
