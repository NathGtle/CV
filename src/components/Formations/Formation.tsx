import React from 'react'
import PropTypes from 'prop-types'

import './Formation.scss'

interface FormationInterface {
  formation: {
    nom: string
    dateDebut: string
    dateFin: string
    description: string
  }
}

const Formation = ({ formation }: FormationInterface) => {
  if (!formation) return null

  return (
    <article className="formation">
      <div className="headline">{`${formation.nom}`}</div>
      <div className="duration">
        {`${formation.dateDebut} ${formation.dateFin ? `- ${formation.dateFin}` : ''}`}
      </div>
      <div className="description">{formation.description}</div>
    </article>
  )
}

Formation.defaultProps = {
  formation: undefined,
}

Formation.propTypes = {
  formation: PropTypes.shape({
    nom: PropTypes.string,
    dateDebut: PropTypes.string,
    dateFin: PropTypes.string,
    description: PropTypes.string,
  }),
}

export default Formation
