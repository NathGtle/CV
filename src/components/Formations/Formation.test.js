import React from 'react'
import _cloneDeep from 'lodash/cloneDeep'
import renderer from 'react-test-renderer'
import Formation from './Formation'

const mocks = {
  formation: {
    nom: 'Docker for entreprise',
    description: 'Formation officiel Docker US',
    dateDebut: 'Avril 2017',
  },
}

describe('Formation component', () => {
  it('should render correctly with props', () => {
    const component = renderer.create(<Formation {...mocks} />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly without props', () => {
    const component = renderer.create(<Formation />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with endate', () => {
    const clonedMocks = _cloneDeep(mocks)
    clonedMocks.formation.dateFin = 'Janvier 2019'
    const component = renderer.create(<Formation {...clonedMocks} />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
