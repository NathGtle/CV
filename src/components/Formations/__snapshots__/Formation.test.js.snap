// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formation component should render correctly with endate 1`] = `
<article
  className="formation"
>
  <div
    className="headline"
  >
    Docker for entreprise
  </div>
  <div
    className="duration"
  >
    Avril 2017 - Janvier 2019
  </div>
  <div
    className="description"
  >
    Formation officiel Docker US
  </div>
</article>
`;

exports[`Formation component should render correctly with props 1`] = `
<article
  className="formation"
>
  <div
    className="headline"
  >
    Docker for entreprise
  </div>
  <div
    className="duration"
  >
    Avril 2017 
  </div>
  <div
    className="description"
  >
    Formation officiel Docker US
  </div>
</article>
`;

exports[`Formation component should render correctly without props 1`] = `null`;
